libafl_frida/harness.cpp

```diff
@@ -0,0 +1,33 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <string>
+
+extern "C" int heap_uaf_read(const uint8_t *_data, size_t _size) {
+  int *array = new int[100];
```

Comment:
are you purposely using c++ allocations instead of raw `malloc`s?

Reply:
Nothing special; more tests should be added that will cover C. I just started from code present in some other example, and it used C++.

Reply:
Ok. So we should make sure to include versions of the tests which use the C api.

Reply:
also, are you certain your tests aren't being optimized out?

Reply:
Added several malloc-based tests and -O0 to the compilation options to make sure the code is not optimized out. We still need a way to run all the tests. Until AsanRuntime cleans up when dropped this is not possible

Reply:
You can go ahead and try to implement the AsanRuntime cleanup. Perhaps as a separate PR?

Reply:
If you run cargo test this one will run. It is already like that

Reply:
FWIW we can also add external tests in a `Makefile.toml` and run these with `cargo make test` (and CI supports it too).
Of course it's nice like it's now, but not everything may be possible that way, so mentioning it as well

Reply:
So, can we resolve it as it is now?